### DISCUSSION TITLE
source does not seem to support multiple files.

### DISCUSSION THREAD

**Discussion by jetersen** [marlonrichert/zsh-snap#32] [Q&A] [answered]
logically I wanted to specify what to source from repo, so was hoping to do
```
znap source asdf asdf.sh completions/_asdf
```
which does not work.

however
```
znap source asdf
znap source asdf completions
```
does work

**Comment by marlonrichert** [comment]
I’m not sure what you mean. Can you please give me a concrete example that I can test with? Which plugin/repo are trying to use and what are you trying to do with it?

**Comment by jetersen** [comment]
I am attempting to source https://github.com/asdf-vm/asdf 's `asdf.sh` file and `completions/_asdf`

**Comment by marlonrichert** [comment, marked as answer]
`znap source asdf asdf.sh` works fine for me.

`completions/_asdf` is a function. You shouldn't `source` functions in Zsh (or any other shell I know of, for that matter); it won't work. `source` is for running scripts only.

Instead, for `_asdf` to be available to Zsh's completion system, you need to add its parent dir to your `$fpath`, like this:
```zsh
fpath+=(  ~[asdf-vm/asdf]/completions )
```
